pulser/simulation/simulation.py

```diff
@@ -219,28 +359,29 @@ def _build_operator(self, op_id: str, *qubit_ids: Union[str, int],
                    else self.op_matrix['I'] for j in range(self._size)]
         return qutip.tensor(op_list)
 
-    def _construct_hamiltonian(self) -> None:
-        def adapt(full_array: np.ndarray) -> np.ndarray:
-            """Adapts list to correspond to sampling rate."""
-            indexes = np.linspace(0, self._tot_duration-1,
-                                  int(self.sampling_rate*self._tot_duration),
-                                  dtype=int)
-            return cast(np.ndarray, full_array[indexes])
+    def _adapt_to_sampling_rate(self, full_array: np.ndarray) -> np.ndarray:
+        """Adapt list to correspond to sampling rate."""
+        indices = np.linspace(0, self._tot_duration-1,
+                              int(self._sampling_rate * self._tot_duration),
+                              dtype=int)
+        return cast(np.ndarray, full_array[indices])
 
-        def make_vdw_term() -> float:
+    def _construct_hamiltonian(self) -> None:
+        def make_vdw_term() -> qutip.Qobj:
             """Construct the Van der Waals interaction Term.
 
             For each pair of qubits, calculate the distance between them, then
             assign the local operator "sigma_rr" at each pair. The units are
             given so that the coefficient includes a 1/hbar factor.
             """
-            vdw = 0
+            vdw = 0 * self._build_operator('I')
```

Comment:
What does this do?

Reply:
Not much ! It was just useful to enforce typing for `vdw`as a `qutip.Qobj`.